### ISSUE TITLE
Error in toConstructor Injection

### ISSUE THREAD

**Issue by pdjohn** [ray-di/Ray.Di#204]
Hello,
I am unable to work constructor injection. Here is my code
```
<?php

declare(strict_types=1);

namespace App\action;

require __DIR__ . '../../../vendor/autoload.php';

use Ray\Di\AbstractModule;
use Ray\Di\Injector;


interface DbInterface
{
}

class Db implements DbInterface
{
    private $dsn;
    private $username;
    private $password;
    public function __construct($dsn, $username, $password)
    {
        $this->dsn = $dsn;
        $this->username = $username;
        $this->password = $password;
    }

}

class FinderModule extends AbstractModule
{
    protected function configure()
    {
        $this->bind(DbInterface::class)->toConstructor(Db::class, [['dsn'=>'dsn'],['username'=>'username'],['password'=>'password']]);
        $this->bind()->annotatedWith('dsn')->toInstance('msql:host=localhost;dbname=test');
        $this->bind()->annotatedWith('username')->toInstance('root');
        $this->bind()->annotatedWith('password')->toInstance('52');

    }
}

$injector = new Injector(new FinderModule);
$temp = $injector->getInstance(Db::class);


print_r($temp);
```

Here is the error report
```PHP Notice:  Array to string conversion in D:\Learning\DITest\vendor\ray\di\src\Bind.php on line 178

Notice: Array to string conversion in D:\Learning\DITest\vendor\ray\di\src\Bind.php on line 178
PHP Notice:  Array to string conversion in D:\Learning\DITest\vendor\ray\di\src\Bind.php on line 178

Notice: Array to string conversion in D:\Learning\DITest\vendor\ray\di\src\Bind.php on line 178
PHP Notice:  Array to string conversion in D:\Learning\DITest\vendor\ray\di\src\Bind.php on line 178

Notice: Array to string conversion in D:\Learning\DITest\vendor\ray\di\src\Bind.php on line 178
PHP Fatal error:  Uncaught exception 'Ray\Di\Exception\Unbound' with message '-'
- dependency '' with name '' used in D:\Learning\DITest\app\action\FinderModule.php:22 ($dsn)

#0 D:\Learning\DITest\vendor\ray\di\src\Container.php(82): Ray\Di\Container->unbound('-')
#1 D:\Learning\DITest\vendor\ray\di\src\Arguments.php(49): Ray\Di\Container->getDependency('-')
#2 D:\Learning\DITest\vendor\ray\di\src\Arguments.php(36): Ray\Di\Arguments->getParameter(Object(Ray\Di\Container), Object(Ray\Di\Argument))
#3 D:\Learning\DITest\vendor\ray\di\src\NewInstance.php(50): Ray\Di\Arguments->inject(Object(Ray\Di\Container))
#4 D:\Learning\DITest\vendor\ray\di\src\Dependency.php(81): Ray\Di\NewInstance->__invoke(Object(Ray\Di\Container))
#5 D:\Learning\DITest\vendor\ray\di\src\Container.php(86): Ray\Di\Dependency->inject(Object(Ray\Di\Container))
#6 D:\Learning\DITest\vendor\ray\di\src\Container.php(52): Ray\Di\Container->getDependency('App\\action\\Db-')
#7 D:\Learning\DITest\vendor\ray\di\src\Injector.php(75): Ray\Di\Container->getInstance('A in D:\Learning\DITest\vendor\ray\di\src\Arguments.php on line 5
6

Fatal error: Uncaught exception 'Ray\Di\Exception\Unbound' with message '-'
- dependency '' with name '' used in D:\Learning\DITest\app\action\FinderModule.php:22 ($dsn)

#0 D:\Learning\DITest\vendor\ray\di\src\Container.php(82): Ray\Di\Container->unbound('-')
#1 D:\Learning\DITest\vendor\ray\di\src\Arguments.php(49): Ray\Di\Container->getDependency('-')
#2 D:\Learning\DITest\vendor\ray\di\src\Arguments.php(36): Ray\Di\Arguments->getParameter(Object(Ray\Di\Container), Object(Ray\Di\Argument))
#3 D:\Learning\DITest\vendor\ray\di\src\NewInstance.php(50): Ray\Di\Arguments->inject(Object(Ray\Di\Container))
#4 D:\Learning\DITest\vendor\ray\di\src\Dependency.php(81): Ray\Di\NewInstance->__invoke(Object(Ray\Di\Container))
#5 D:\Learning\DITest\vendor\ray\di\src\Container.php(86): Ray\Di\Dependency->inject(Object(Ray\Di\Container))
#6 D:\Learning\DITest\vendor\ray\di\src\Container.php(52): Ray\Di\Container->getDependency('App\\action\\Db-')
#7 D:\Learning\DITest\vendor\ray\di\src\Injector.php(75): Ray\Di\Container->getInstance('A in D:\Learning\DITest\vendor\ray\di\src\Arguments.php on line 5
6
```

I am using 
`"ray/di": "^2.10"`
`"php": "^7.2.5"`

Can you please look into it?

Thanks.

**Comment by koriym** [comment]
`Unbound` exception raised because `Db` **is** unbound. You bound `DbInterface` with `toConstructor`,  So please  get an instance with `DbIntetface` as following.

```php
$temp = $injector->getInstance(DbInterface::class);
```

**Comment by pdjohn** [comment]
After getting the `DbInterface` as you said, I am still getting the following error report

```
PHP Notice:  Array to string conversion in D:\Learning\DITest\vendor\ray\di\src\Bind.php on line 178

Notice: Array to string conversion in D:\Learning\DITest\vendor\ray\di\src\Bind.php on line 178
PHP Notice:  Array to string conversion in D:\Learning\DITest\vendor\ray\di\src\Bind.php on line 178

Notice: Array to string conversion in D:\Learning\DITest\vendor\ray\di\src\Bind.php on line 178
PHP Notice:  Array to string conversion in D:\Learning\DITest\vendor\ray\di\src\Bind.php on line 178

Notice: Array to string conversion in D:\Learning\DITest\vendor\ray\di\src\Bind.php on line 178
PHP Fatal error:  Uncaught exception 'Ray\Di\Exception\Unbound' with message '-'
- dependency '' with name '' used in D:\Learning\DITest\app\action\FinderModule.php:23 ($dsn)

#0 D:\Learning\DITest\vendor\ray\di\src\Container.php(82): Ray\Di\Container->unbound('-')
#1 D:\Learning\DITest\vendor\ray\di\src\Arguments.php(49): Ray\Di\Container->getDependency('-')
#2 D:\Learning\DITest\vendor\ray\di\src\Arguments.php(36): Ray\Di\Arguments->getParameter(Object(Ray\Di\Container), Object(Ray\Di\Argument))
#3 D:\Learning\DITest\vendor\ray\di\src\NewInstance.php(50): Ray\Di\Arguments->inject(Object(Ray\Di\Container))
#4 D:\Learning\DITest\vendor\ray\di\src\Dependency.php(81): Ray\Di\NewInstance->__invoke(Object(Ray\Di\Container))
#5 D:\Learning\DITest\vendor\ray\di\src\Container.php(86): Ray\Di\Dependency->inject(Object(Ray\Di\Container))
#6 D:\Learning\DITest\vendor\ray\di\src\Container.php(52): Ray\Di\Container->getDependency('App\\action\\DbIn...')
#7 D:\Learning\DITest\vendor\ray\di\src\Injector.php(60): Ray\Di\Container->getInstanc in D:\Learning\DITest\vendor\ray\di\src\Arguments.php on line 56

Fatal error: Uncaught exception 'Ray\Di\Exception\Unbound' with message '-'
- dependency '' with name '' used in D:\Learning\DITest\app\action\FinderModule.php:23 ($dsn)

#0 D:\Learning\DITest\vendor\ray\di\src\Container.php(82): Ray\Di\Container->unbound('-')
#1 D:\Learning\DITest\vendor\ray\di\src\Arguments.php(49): Ray\Di\Container->getDependency('-')
#2 D:\Learning\DITest\vendor\ray\di\src\Arguments.php(36): Ray\Di\Arguments->getParameter(Object(Ray\Di\Container), Object(Ray\Di\Argument))
#3 D:\Learning\DITest\vendor\ray\di\src\NewInstance.php(50): Ray\Di\Arguments->inject(Object(Ray\Di\Container))
#4 D:\Learning\DITest\vendor\ray\di\src\Dependency.php(81): Ray\Di\NewInstance->__invoke(Object(Ray\Di\Container))
#5 D:\Learning\DITest\vendor\ray\di\src\Container.php(86): Ray\Di\Dependency->inject(Object(Ray\Di\Container))
#6 D:\Learning\DITest\vendor\ray\di\src\Container.php(52): Ray\Di\Container->getDependency('App\\action\\DbIn...')
#7 D:\Learning\DITest\vendor\ray\di\src\Injector.php(60): Ray\Di\Container->getInstanc in D:\Learning\DITest\vendor\ray\di\src\Arguments.php on line 56
```

**Comment by koriym** [comment]
Got it.

README was wrong.
Please change `toConstructor` bind as a following.

NG
```php
 $this->bind(DbInterface::class)->toConstructor(Db::class, [['dsn'=>'dsn'],['username'=>'username'],['password'=>'password']]);
```

OK
```php
 $this->bind(DbInterface::class)->toConstructor(Db::class, ['dsn'=>'dsn', 'username'=>'username','password'=>'password']);
```

This should be work. 

Now we're aware that we should provide better error message. I should work on it.

**Comment by koriym** [comment]
README fixed and tests added in #205.

@pdjohn Sorry for the inconvenience and thank you for the reporting the issue!
